ci/Dockerfile-envoy-image

```diff
@@ -1,26 +1,20 @@
 FROM ubuntu:16.04
 
-ARG dumbinit_version=1.2.1
-ENV loglevel=info
-
 RUN apt-get update \
     && apt-get upgrade -y \
-    && apt-get install -y wget ca-certificates \
+    && apt-get install -y ca-certificates \
     && apt-get autoremove -y \
     && apt-get clean \
     && rm -rf /tmp/* /var/tmp/* \
     && rm -rf /var/lib/apt/lists/*
 
-
-RUN wget https://github.com/Yelp/dumb-init/releases/download/v${dumbinit_version}/dumb-init_${dumbinit_version}_amd64.deb \
-    && dpkg -i dumb-init_${dumbinit_version}_amd64.deb
-
 RUN mkdir -p /etc/envoy
 
 ADD build_release_stripped/envoy /usr/local/bin/envoy
 ADD configs/google_com_proxy.v2.yaml /etc/envoy/envoy.yaml
 
 EXPOSE 10000
 
-ENTRYPOINT ["/usr/bin/dumb-init", "--"]
-CMD /usr/local/bin/envoy --v2-config-only -l $loglevel -c /etc/envoy/envoy.yaml
+COPY docker-entrypoint.sh /
+ENTRYPOINT ["/docker-entrypoint.sh"]
+CMD ["envoy", "--v2-config-only", "-c", "/etc/envoy/envoy.yaml"]
```

Comment:
I kind of wish we had a better CI environment where we could just run the result Docker images with a bunch of different CLI args to validate behavior, CC @klarose who was looking into remote executors for CircleCI.

Reply:
The remote executor works pretty well if you don't need a beefy machine. In my case, I compiled the test binaries locally on the large machine we normally use, then had docker run them on the remote machine. I did this by copying the binaries into a new layer on the container before doing the run.

We should be able to do something similar here -- build the docker images locally on the large machine, then tell the remote to run them. Since it's only an envoy doing pretty much nothing, the size of the machine shouldn't be a problem (I hope!).

Once we have that in place, we can just have a script which executes a series of docker run commands with the arguments we want, comparing the results against the output we expect.